Backend/src/controllers/graph.controller.ts

```diff
@@ -142,144 +153,507 @@ const getClusterProperties = async (req: Request, res: Response) => {
+        const result = running.map((dbRow) => ({
+            userId: dbRow.user_id,
+            graphId: dbRow.graph_id,
+            hdfsIp: dbRow.hdfs_ip,
+            hdfsPort: dbRow.hdfs_port,
+            hdfsFilePath: dbRow.hdfs_file_path,
+            llmRunnerString: dbRow.llm_runner_string,
+            inferenceEngine: dbRow.inference_engine,
+            model: dbRow.model,
+            chunkSize: dbRow.chunk_size,
+            status: dbRow.status,
+            message: dbRow.message,
+            clusterId: dbRow.cluster_id,
+        }));
+
+        return res.status(200).json({ data: result });
+    } catch (err) {
+        console.error(err);
+        return res.status(500).json({
+            message:
+                "Internal Server Error: Unable to fetch KGConstructionMeta for the given cluster and file path.",
+            error: err instanceof Error ? err.message : "Unknown error occurred",
+        });
+    }
+};
 
-      // Write the command to the Telnet server
-      tSocket.write(GRAPH_UPLOAD_COMMAND + '|' + graphName + '|' + filePath + '\n', "utf8", () => {
-        setTimeout(() => {
-          if (commandOutput) {
-            console.log(new Date().toLocaleString() + " - UPLOAD " + req.body.graphName + " - " + commandOutput);
-            res.status(HTTP[200]).send(commandOutput);
-          } else {
-            res.status(HTTP[400]).send({ code: ErrorCode.NoResponseFromServer, message: ErrorMsg.NoResponseFromServer, errorDetails: ErrorMsg.NoResponseFromServer });
-          }
-        }, TIMEOUT.hundred); // Adjust timeout to wait for the server response if needed
-      });
-    });
-  } catch (err) {
-    return res.status(HTTP[200]).send({ code: ErrorCode.ServerError, message: ErrorMsg.ServerError, errorDetails: err });
-  }
+export const updateKGConstructionMetaByClusterId = async (
+    req: Request,
+    res: Response
+) => {
+    const { clusterId, hdfsFilePath } = req.params;
+    const updateData = req.body;
+
+    try {
+        const metaData = await getKGConstructionMetaByClusterRepo(Number(clusterId));
+        const target = metaData.find((m) => m.hdfs_file_path === hdfsFilePath);
+
+        if (!target) {
+            return res.status(404).json({
+                message: `No KG construction metadata found for clusterId: ${clusterId} and hdfsFilePath: ${hdfsFilePath}`,
+            });
+        }
+
+        const updated = await updateKGConstructionMetaStatusRepo(
+            target.id,
+            updateData.status as KGStatus,
+            updateData.message
+        );
+
+        return res.status(200).json({
+            message: "KGConstructionMeta updated successfully",
+            data: updated,
+        });
+    } catch (err) {
+        console.error(err);
+        return res.status(500).json({
+            message:
+                "Internal Server Error: Unable to update KGConstructionMeta for the given cluster and file path.",
```

Comment:
What do you mean by `KGConstructionMeta`? Are you referring to `KG Construction Metadata`?

Reply:
Ok